File: 백준/Bronze/2884. 알람 시계/알람 시계.py
H, M = map(int, input().split())

if(M >= 45):
    M -= 45
else:
    if(H == 0):
        H = H - 1 + 24
    else:
        H -= 1
    M = M - 45 + 60
print(H, M)

File: 백준/Bronze/2884. 알람 시계/알람 시계.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split(' ');
let h = parseInt(input[0]);
let m = parseInt(input[1]);

if (m >= 45) {
    m -= 45;
} else {
    if (h === 0) {
        h = 24 - 1;
    } else {
        h -= 1;
    }
    m = m - 45 + 60;
}

console.log(`${h} ${m}`);